tools/parse/src/main/scala/org/allenai/nlpstack/parse/poly/decisiontree/RandomForest.scala

```diff
@@ -80,36 +86,38 @@ class RandomForestTrainer(validationPercentage: Double, numDecisionTrees: Int,
   maximumDepthPerTree: Int = Integer.MAX_VALUE)
     extends ProbabilisticClassifierTrainer {
 
+  private val dtTrainer = new DecisionTreeTrainer(validationPercentage, featuresExaminedPerNode,
+    maximumDepth = maximumDepthPerTree)
+
   /** Induces a RandomForest from a set of feature vectors.
     *
     * @param data a set of feature vectors to use for training
     * @return the induced random forest
     */
   override def apply(data: FeatureVectorSource): ProbabilisticClassifier = {
```

Comment:
Do you know about the internal structure of the trees?

In Java, when you read a document into a string, and then get a a single word out of the document with `String.substring()`, it will hold the entire document in memory until you release that word, because the `String` you get back from `String.substring()` holds a reference to the string it came from. You can get around it by saying `String standaloneString = new String(myWord)`.

I wonder if a similar thing is going on with these DTs, and when you serialize and then deserialize, it loses that reference to the original training data structures?